BDArmory/FX/ExplosionFX.cs

```diff
@@ -628,28 +644,45 @@ private void ExecuteBuildingBlastEvent(BuildingBlastHitEvent eventToExecute)
             //TODO: Review if the damage is sensible after so many changes
             //buildings
             DestructibleBuilding building = eventToExecute.Building;
-            building.damageDecay = 600f;
+            //building.damageDecay = 600f;
 
-            if (building)
+            if(building && building.IsIntact)
             {
                 var distanceFactor = Mathf.Clamp01((Range - eventToExecute.Distance) / Range);
-                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW * Power * distanceFactor;
-
-                damageToBuilding *= 2f;
+                float blastMod = 1;
+                switch (ExplosionSource)
+                {
+                    case ExplosionSourceType.Bullet:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW;
+                        break;
+                    case ExplosionSourceType.Rocket:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_ROCKET;
+                        break;
+                    case ExplosionSourceType.Missile:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_MISSILE;
+                        break;
+                    case ExplosionSourceType.BattleDamage:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BATTLE_DAMAGE;
+                        break;
+                }
+                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * blastMod * (Power * distanceFactor);
+                damageToBuilding /= 2;
                 damageToBuilding *= BDArmorySettings.BUILDING_DMG_MULTIPLIER;
+                //building.AddDamage(damageToBuilding); 
+                building.FacilityDamageFraction += damageToBuilding;
 
-                building.AddDamage(damageToBuilding);
-
-                if (building.Damage > building.impactMomentumThreshold)
+                if (building.FacilityDamageFraction > building.impactMomentumThreshold * 2)
```

Comment:
impactMomentumThreshold varies from building to building, even when unmodifed by BDArmory.Damage.BuildingDamage, so it works well as a HP limit